### DISCUSSION TITLE
Feature Request: `routeMeta` function

### DISCUSSION THREAD

**Discussion by knpwrs** [solidjs/solid-start#537] [Ideas]
Inspired by Remix's `export meta` feature: https://remix.run/docs/en/v1/route/meta

Essentially you can export a `meta` function from route components, and then higher up in the component tree use [`useMatches`](https://remix.run/docs/en/v1/hooks/use-matches) to get all of the matching `meta` functions and use the data.

Two example use cases come to mind:

1. Breadcrumbs (example on [this page](https://remix.run/docs/en/v1/hooks/use-matches)). Route component files can return `title` and `href` properties from their `meta` functions. A parent component can get all of these results and render breadcrumbs.
2. Shared layout titles. Similar to breadcrumbs, basically just getting a title for whatever is being rendered inside of `<Outlet />` and rendering it in a consistent way for all child routes.

**Comment by semiautomatix** [comment]
This would really help interpreting breadcrumbs.

**Comment by ryansolid** [comment]
I've added `info` to the RouteDefinition in Solid router.. you can set it through `export const route = {}` so I hope this addresses this need.

**Reply by knpwrs** [reply]
Are there any examples of how to use `info` on `RouteDefinition`?

**Comment by blopker** [comment]
I've dug around to see how this might work. I'll share what I've learned so far in case it helps anyone. Sadly, I can't figure out why this doesn't work with `mdx` files. The `mdx` compiler seems to mangle exports in a way that Solid Router can't pick up.

Say we're doing a blog, and this is a post:
`src/routes/(blog)/the-best-post.tsx`
```tsx
export const route = {
  info: {
    title: "Best Blog Post",
    description: "Maybe",
  },
};
export default function BestPost(props: any) {
  return <div>Latin, or something</div>;
}
```

And here's the layout file:
`src/routes/(blog).tsx`
```tsx
import type { RouteSectionProps } from "@solidjs/router";
import { useCurrentMatches } from "@solidjs/router";
export default function BlogLayout(props: RouteSectionProps) {
  const matches = useCurrentMatches();
  // Roll up all the info objects from the matched routes into a single object
  const info = matches.reduce((acc, m) => {
    return { ...acc, ...m.route.info };
  }, {}) as any;
  return (
    <div>
      <h1>Posts</h1>
      <h2>{info.title}</h2>
      <p>{info.description}</p>
      {props.children}
    </div>
  );
}
```
This works. The reduce will roll up the `info` objects from rested routes in the order of least specific to most into one info object, which is nice. However, I'd expect this to be pretty common for people to do, so it could be more ergonomic. It also lacks type safety, and likely is not reactive.

I don't understand why when using `solid-mdx` this doesn't work, but the compiled files seem to indicate the router never sees the `route` object. I don't understand the mechanism around how the non-mdx export is picked up yet.
